### ISSUE TITLE
[Bug] XF 5.0.0.2012 breaks TitleView width in iOS

### ISSUE THREAD

**Issue by Qythyx** [xamarin/Xamarin.Forms#13795]
### Description
This problem is iOS only, Android is fine. In Xamarin Forms 5.0.0.2012 the width of the TitleView is incorrect. It seems to no longer take into account the hamburger menu. This means that the width is wider than it should be, and anything that is aligned to the right, or full-width, will extend off the right side of the screen.

### Steps to Reproduce

1. Set a Shell.SetTitleView with a view that is either End-aligned or full-width.

### Expected Behavior
Should not extend off the right side of the screen.

### Actual Behavior
Extends off the right side of the screen.

### Basic Information

- Version with issue: 5.0.0.2012
- Last known good version: previous 5.0 version, I forget the exact number

### Screenshots
![image](https://user-images.githubusercontent.com/487802/108351075-5d081080-7228-11eb-8df2-176e2677abaf.png)

### Reproduction Link
This is what I'm placing in the TitleView:

```
<?xml version="1.0" encoding="UTF-8" ?>
<ContentView
	x:Class="BeerApp.App.Forms.Controls.TitleView"
	xmlns="http://xamarin.com/schemas/2014/forms"
	xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml">
	<ContentView.Content>
		<Grid
			ColumnDefinitions="Auto,Auto"
			ColumnSpacing="0"
			HorizontalOptions="End"
			RowDefinitions="*,2"
			RowSpacing="0">
			<ImageButton
				Grid.Row="0"
				Grid.Column="0"
				Command="{Binding GotoOffers}"
				Source="icon_offers.png" />
			<ImageButton
				Grid.Row="0"
				Grid.Column="1"
				Command="{Binding GotoSummary}"
				Source="icon_beerbox.png" />
			<BoxView
				Grid.Row="1"
				Grid.Column="0"
				BackgroundColor="{AppThemeBinding Light={StaticResource LightColorPrimary},
												  Dark={StaticResource DarkColorPrimary}}"
				HeightRequest="2"
				IsVisible="{Binding IsOffers}" />
			<BoxView
				Grid.Row="1"
				Grid.Column="1"
				BackgroundColor="{AppThemeBinding Light={StaticResource LightColorPrimary},
												  Dark={StaticResource DarkColorPrimary}}"
				HeightRequest="2"
				IsVisible="{Binding IsSummary}" />
		</Grid>
	</ContentView.Content>
</ContentView>
```

### Workaround
None.

**Comment by Qythyx** [comment]
To clarify, if I downgrade my iOS project's reference to Xamarin.Forms 5.0.0.1931 then this renders correctly like this:
![image](https://user-images.githubusercontent.com/487802/108352286-f8e64c00-7229-11eb-9644-6e96869a1a9c.png)

**Comment by MrtnSys** [comment]
Hello everyone. I want to use the Shell FlyoutFooter to display some icons in the flyout. Therefore I updated to Xamarin 5.0.0.1931 and then had to find out that the Shell TitleView is broken on iOS. I reverted to Xamarin 4.8.0.1821 and waited for the fix in [https://github.com/xamarin/Xamarin.Forms/pull/13514](url). And now in Xamarin 5.0.0.2012 this bug on iOS and the title is still broken. Oh man. I don't want to tell my customer he has to wait again. I try to add some negative margin to the title container to the left, to get my buttons on the right back.

**Comment by n0jo4** [comment]
It seems like the Shell Menu button width is not being used in the calculation for the width of the TitleView, my workaround is adding a iOS platform specific right margin of 80 to the right most control and it's reasonably close on the devices I've tested.

**Comment by eddieyanez** [comment]
I also have this problem. I was forced to update Xamarin Forms otherwise I can no longer use Hot Reload on the Mac.

**Comment by MrtnSys** [comment]
Use the workaround to add a specific margin to the right in your TitleView-Template Control.

from scratch.. See "OnPlatform iOS" and margin of 60 to the right. That will bring your buttons back. Remove the margin when Xamarin is fixed.

```
<ContentView x:Name=titleView>
    <ContentView.Content>
            <StackLayout
                Orientation="Horizontal"
                HorizontalOptions="EndAndExpand" 
                Margin="{OnPlatform iOS='0,0,60,0', Default='0,0,10,0'}">
		  <Button Text="1" />
		  <Button Text="2" />
		  <Button Text="3" />
	    </StackLayout>
    </ContentView.Content>
</ContentView>
```

**Comment by Qythyx** [comment]
@MrtnSys , thanks for the workaround. I'm curious about that OnPlatform syntax. I can't find it documented anywhere and it doesn't seem to work for me. I'm using this instead, but yours is more compact. How can I use it?
```c#
	<Grid.Margin>
		<OnPlatform x:TypeArguments="Thickness">
			<On Platform="iOS" Value="0,0,60,0" />
			<On Platform="Default" Value="0,0,0,0" />
		</OnPlatform>
	</Grid.Margin>
```

**Comment by AndreasReitberger** [comment]
@PureWeen 
Here is also a good example showing the issue with the TitleView in the latest SR3.

**Comment by PureWeen** [comment]
If anyone wants to test the nugets from my PR

https://dev.azure.com/xamarin/public/_build/results?buildId=40567&view=artifacts&pathAsName=false&type=publishedArtifacts

Let me know if it resolves the issue for you

**Comment by NPadrutt** [comment]
@PureWeen is there a service release upcoming that deploys this?

**Comment by Qythyx** [comment]
It looks like this has been fixed in XF 5.0.0.2083. Woohoo!
